packages/react-router-dom/server.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {
+  Action,
+  Path,
+  LocationPieces,
+  To,
+  createPath,
+  parsePath
+} from 'history';
+import { Router } from 'react-router-dom';
+
+/**
+ * A <Router> that may not transition to any other location. This is useful
+ * on the server where there is no stateful UI.
+ */
+export function StaticRouter({
+  children,
+  location: loc = '/'
+}: StaticRouterProps) {
+  if (typeof loc === 'string') {
+    loc = parsePath(loc);
+  }
+
+  let action = Action.Pop;
+  let location = {
+    pathname: loc.pathname || '/',
+    search: loc.search || '',
+    hash: loc.hash || '',
+    state: loc.state || null,
+    key: loc.key || 'default'
+  };
```

Comment:
Possibly extracting that to function like `getLocationOrDefault`?

Reply:
I'd prefer to leave it inline for now, thanks.